### ISSUE TITLE
SampleData.json does not work together with Gorilla SDK

### ISSUE THREAD

**Issue by escamoteur** [UXDivers/Gorilla-Player-Support#55]
Hi,

after first trying Gorilla Player it worked great with my SampleData.json.

Now I wanted to see my CustomRenderer and I integrated the SDK into my App. Works great after finding the improtant information in: https://github.com/UXDivers/Gorilla-Player-Support-And-Issues/issues/20#issuecomment-172488525

You should at least include this link in your documentation.

Anyway. What doesn't work anymore is my SampleData.json. My ListView stays empty.
What do I miss?

Best
Thomas

**Comment by LeoHere** [comment]
Hi Thomas!

Sample data should work your as expected with the SDK. Couple of things to check:
- The SampleData.json should be placed at the root of the project where the Xaml you want to preview is.
- Build action should be None.

If those two are right, please provide me as much info as you can from your solution/project structure.
- Where is the XAML with the ListView and the SampleData.json? What kind of project are you using?
- Can you include the XAML file and JSON?

Thanks,
Leo

PD: Regarding the SDK, we are working on it. We are simplifying the process of including the SDK, so it is less error prone, adding the derived classes stuff and improving the documentation too.

**Comment by escamoteur** [comment]
Hi Leo,

here is my project structure:



I tried both, putting SampleData.JSON in the PCL and in the Android 
Project.

I attach the JSON file. Th xaml-file has been modified so the binding is 
no longer as it's needed for your Template solution. But it was working 
when using the standard GorillaPlayer.

Good Luck
Thomas
------ Originalnachricht ------
Von: "Leonardo Rodríguez" <notifications@github.com>
An: "UXDivers/Gorilla-Player-Support-And-Issues" 
<Gorilla-Player-Support-And-Issues@noreply.github.com>
Cc: "escamoteur" <thomas@burkharts.net>
Gesendet: 09.03.2016 09:58:53
Betreff: Re: [Gorilla-Player-Support-And-Issues] SampleData.json does 
not work together with Gorilla SDK (#55)

**Comment by LeoHere** [comment]
I can't see neither the image with the project structure nor the SampleData.JSON you mentioned in your post (BTW, is that the name of the file or is SampleData.json? the upper case could be an issue too)

**Comment by escamoteur** [comment]
Sorry, I was not aware that the Email is from GITHUB

![image](https://cloud.githubusercontent.com/assets/16635729/13668935/53a0014a-e6c0-11e5-8c89-19fde2d9958d.png)

[SampleData.zip](https://github.com/UXDivers/Gorilla-Player-Support-And-Issues/files/167070/SampleData.zip)

**Comment by LeoHere** [comment]
Ok. The name of the page in the SampleData.json seems to be wrong. In the .json you have "Page1.xaml" meanwhile the xaml file is DayPage.xaml.

Let me know if that fixes the issue.

**Comment by escamoteur** [comment]
Hi Leo,

I just checked again the SampleData.json I attached in the zip-File there is no Page1.xaml:

```
{
  "DayPage.xaml": [

    {"Quote" : "TestQuote", "DateAsString" : "29.12.2015" }
  ]
}
```

**Comment by LeoHere** [comment]
Yep, you are right. Too many SampleData.json in my downloads folder, time to do some cleaning. Sorry for that.

I don't see anything wrong. 

Can you provide a sample solution to reproduce the issue? If you prefer using email let me know.

**Comment by escamoteur** [comment]
I'm sorry that I can't provide you the project. Would the log-files help you?

**Comment by LeoHere** [comment]
Ok. No problem. Yes logs could help, but we will need debug logs.

In order to generate debug you need to:
1) Stop Gorilla. In the Tray icon, click on Quit.
2) Open config.json located  "Users\<YOUR_USER>\.gorilla\config.json" (Windows) or "~\.gorilla\config.json" (mac)
3) You should have something like {"serverName":"Gorilla on <your computer name>"}
4) Add logLevel debug, like this {"serverName":"Gorilla on <your computer name>", "logLevel":"debug"}
5) Save it
6) Launch Gorilla host app again
7) Try to preview the DayPage.xaml
8) Go to Users Users\<YOUR_USER>\.gorilla\logs (Windows) or ~\.gorilla\logs (mac) and send me the log files
9) If possible also attach the DayPage.xaml
10) (optional) undo the change done in 4)

**Comment by LeoHere** [comment]
Closing the issue. If you are still experiencing the issue please open a new issue referencing this one.

**Comment by iforlife** [comment]
In my case it was some invalid character in the json that I needed to escape. Once I moved the json in the xaml file as a resource it all turned purple because of the invalid character :) I still cannot make it work with the SampleData.json file though - it all needs to be included as a resource.

**Comment by LeoHere** [comment]
@iforlife can you post this issue in a separate thread please? Also please include details about the specific scenario you are trying to preview. Tks

**Comment by jt-github** [comment]
Strangely, I had this same issue.  I tried naming my json file the old way (SampleData.json) and the new way (DesignTimeData.json), putting different (very simple) sample data in it, moved it around to different places, and even destroyed and recreated it a few times, and I got the Missing Data error no matter what I did.

I restarted Visual Studio, the Gorilla client, and server multiple times and changed the client logging level multiple times with no improvement (other than seeing more verbose error output, of course).  _Once I changed the **server** logging level to debug, however, my DesignTimeData.json file was suddenly seen by Gorilla and rendered properly._  Thanks, @LeoHere!

I shut down Visual Studio, the client and the server, then set the server back to the default log level and started everything back up again and it's still working.  No idea what that was all about.
